JSTests/wasm/function-references/tail_call.js

```diff
@@ -0,0 +1,117 @@
+//@ runWebAssemblySuite("--useWebAssemblyTypedFunctionReferences=true", "--useWebAssemblyTailCalls=true", "--useWasmLLInt=true", "--forceAllFunctionsToUseSIMD=false")
+//
+// FIXME: The extra LLInt flag is needed because tail calls aren't supported in BBQ yet.
+// We also disable the forceAllFunctionsToUseSIMD flag as it isn't compatible with tail calls either.
+
+import * as assert from '../assert.js';
+import { instantiate } from "../wabt-wrapper.js";
+
+function module(bytes, valid = true) {
+  let buffer = new ArrayBuffer(bytes.length);
+  let view = new Uint8Array(buffer);
+  for (let i = 0; i < bytes.length; ++i) {
+    view[i] = bytes.charCodeAt(i);
+  }
+  return new WebAssembly.Module(buffer);
+}
+
+async function returnCallRef() {
+  /*
+  (module
+    (elem declare funcref (ref.func 0))
+    (func (export "g") (result i32) (i32.const 42))
+    (func (export "f") (param (ref 0)) (result i32)
+      (return_call_ref 0 (local.get 0))
+      (i32.const 5))
+  )
+  */
+  {
+    let m = new WebAssembly.Instance(module(
+      "\x00\x61\x73\x6d\x01\x00\x00\x00\x01\x8b\x80\x80\x80\x00\x02\x60\x00\x01\x7f\x60\x01\x64\x00\x01\x7f\x03\x83\x80\x80\x80\x00\x02\x00\x01\x07\x89\x80\x80\x80\x00\x02\x01\x67\x00\x00\x01\x66\x00\x01\x09\x87\x80\x80\x80\x00\x01\x07\x70\x01\xd2\x00\x0b\x0a\x97\x80\x80\x80\x00\x02\x84\x80\x80\x80\x00\x00\x41\x2a\x0b\x88\x80\x80\x80\x00\x00\x20\x00\x15\x00\x41\x05\x0b"
+    ));
+    assert.eq(m.exports.f(m.exports.g), 42);
+  }
+
+  // FIXME: blocked on https://bugs.webkit.org/show_bug.cgi?id=265262
```

Comment:
There are some tests that still crash because of https://bugs.webkit.org/show_bug.cgi?id=265262 (an existing bug with tail calls), so I've left those commented out.